contracts/multisig/src/contract.rs

```diff
@@ -251,6 +278,15 @@ impl Multisig {
             panic_with_error!(&env, ContractError::ProposalClosed);
         }
 
+        let curr_timestamp = env.ledger().timestamp();
+        if curr_timestamp > proposal.expiration_timestamp {
+            log!(
+                &env,
+                "Multisig: Execute proposal: Trying to execute an expired proposal!"
+            );
+            panic_with_error!(&env, ContractError::ProposalExpired);
+        }
+
```

Comment:
At this point we could also automatically set `proposal_state.ProposalExpired`.

Reply:
added that to the source, but I don't think we can test it currently unless we change the return type to `Result`